modules/tensor_mechanics/src/materials/HillElastoPlasticityStressUpdate.C

```diff
@@ -61,20 +70,108 @@ HillElastoPlasticityStressUpdateTempl<is_ad>::HillElastoPlasticityStressUpdateTe
+                                                                             "hill_tensor")),
+    _rotation_matrix(
+        this->template declareProperty<RankTwoTensor>(this->_base_name + "rotation_matrix")),
+    _rotation_matrix_transpose(this->template declareProperty<RankTwoTensor>(
+        this->_base_name + "rotation_matrix_transpose")),
+    _rotation_matrix_old(
+        this->template getMaterialPropertyOld<RankTwoTensor>(this->_base_name + "rotation_matrix")),
+    _rotation_matrix_transpose_old(this->template getMaterialPropertyOld<RankTwoTensor>(
+        this->_base_name + "rotation_matrix_transpose")),
+    _local_cylindrical_csys(this->template getParam<bool>("local_cylindrical_csys")),
+    _local_spherical_csys(this->template getParam<bool>("local_spherical_csys")),
+    _axis(this->template getParam<int>("axis")),
+    _center(this->template getParam<Point>("center"))
 {
 }
 
+template <bool is_ad>
+void
+HillElastoPlasticityStressUpdateTempl<is_ad>::initQpStatefulProperties()
+{
+  RealVectorValue normal_vector(0, 0, 0);
+  RealVectorValue axial_vector(0, 0, 0);
+
+  if (_local_cylindrical_csys && _local_spherical_csys)
+  {
+    mooseError("\nBoth local_cylindrical_csys and local_spherical_csys were provided. Neither or "
+               "only one can be provided");
+  }
+  if (_local_cylindrical_csys)
+  {
+    if (_axis == 0)
+    {
+      normal_vector(0) = 0.0;
+      normal_vector(1) = _q_point[_qp](1);
+      normal_vector(2) = _q_point[_qp](2);
+      axial_vector(0) = 1.0;
+    }
+    else if (_axis == 1)
+    {
+      normal_vector(0) = _q_point[_qp](0);
+      normal_vector(1) = 0.0;
+      normal_vector(2) = _q_point[_qp](2);
+      axial_vector(1) = 1.0;
+    }
+    else if (_axis == 2)
+    {
+      normal_vector(0) = _q_point[_qp](0);
+      normal_vector(1) = _q_point[_qp](1);
+      normal_vector(2) = 0.0;
+      axial_vector(2) = 1.0;
+    }
+    else
+      mooseError("\nInvalid value for axis parameter provided in HillElastoPlasticityStressUpdate");
+  }
+
+  if (_local_spherical_csys)
+  {
+    normal_vector(0) = _q_point[_qp](0) - _center(0);
+    normal_vector(1) = _q_point[_qp](1) - _center(1);
+    normal_vector(2) = _q_point[_qp](2) - _center(2);
+    axial_vector(1) = 1; // For transverse isotropic sphere axial_vector does not matter but can't
+                         // be zero. Arbitrarily set here.
+  }
+
+  if (_local_cylindrical_csys || _local_spherical_csys)
+  {
+    Real nv_norm = normal_vector.norm();
+    if (!(MooseUtils::absoluteFuzzyEqual(nv_norm, 0.0)))
+      normal_vector /= nv_norm;
+    else
+    {
+      mooseError("The normal vector cannot be a zero vector in "
+                 "HillElastoPlasticityStressUpdateBisonRotated");
+    }
+
+    RankTwoScalarTools::setRotationMatrix(
+        normal_vector, axial_vector, _rotation_matrix[_qp], false);
+    _rotation_matrix_transpose[_qp] = _rotation_matrix[_qp].transpose();
```

Comment:
It can be obtained by transposing; however that transpose operation will need to performed in the newton's loop for every iteration - that may be expensive. Right now this transpose operation is just done once and the _rotation_matrix_transpose is then propagated to the next time step. Let me know if you think otherwise.